allennlp/common/configuration.py

```diff
@@ -0,0 +1,262 @@
+            "\n"
+    ])
+
+    return rendered_item
+
+BASE_CONFIG: Config = Config([
+        ConfigItem(name="dataset_reader",
+                   annotation=DatasetReader,
+                   default_value=_NO_DEFAULT,
+                   comment="specify your dataset reader here"),
+        ConfigItem(name="validation_dataset_reader",
+                   annotation=DatasetReader,
+                   default_value=None,
+                   comment="same as dataset_reader by default"),
+        ConfigItem(name="train_data_path",
+                   annotation=str,
+                   default_value=_NO_DEFAULT,
+                   comment="path to the training data"),
+        ConfigItem(name="validation_data_path",
+                   annotation=str,
+                   default_value=None,
+                   comment="path to the validation data"),
+        ConfigItem(name="test_data_path",
+                   annotation=str,
+                   default_value=None,
+                   comment="path to the test data (you probably don't want to use this!)"),
+        ConfigItem(name="evaluate_on_test",
+                   annotation=bool,
+                   default_value=False,
+                   comment="whether to evaluate on the test dataset at the end of training (don't do it!"),
+        ConfigItem(name="model",
+                   annotation=Model,
+                   default_value=_NO_DEFAULT,
+                   comment="specify your model here"),
+        ConfigItem(name="iterator",
+                   annotation=DataIterator,
+                   default_value=_NO_DEFAULT,
+                   comment="specify your data iterator here"),
+        ConfigItem(name="trainer",
+                   annotation=Trainer,
+                   default_value=_NO_DEFAULT,
+                   comment="specify the trainer parameters here"),
+        ConfigItem(name="datasets_for_vocab_creation",
+                   annotation=List[str],
+                   default_value=None,
+                   comment="if not specified, use all datasets"),
+        ConfigItem(name="vocabulary",
+                   annotation=Vocabulary,
+                   default_value=None,
+                   comment="vocabulary options"),
+
+])
+
+def _valid_choices(cla55: type) -> Dict[str, str]:
+    """
+    Return a mapping {registered_name -> subclass_name}
+    for the registered subclasses of `cla55`.
+    """
+    choices: Dict[str, str] = {}
+
+    if cla55 not in Registrable._registry:
+        raise ValueError(f"{cla55} is not a known Registrable class")
+
+    for name, subclass in Registrable._registry[cla55].items():
+        # These wrapper classes need special treatment
+        if isinstance(subclass, (_Seq2SeqWrapper, _Seq2VecWrapper)):
+            subclass = subclass._module_class
+
+        choices[name] = full_name(subclass)
+
+    return choices
+
+def configure(full_path: str = '') -> Union[Config, List[str]]:
```

Comment:
It would be easy for this to support directories as well - it would be great if passing a directory contained within allennlp resulted in something like:

```
allennlp configure models

Found Registerable "Model" with subclasses:
SemanticRoleLabeller
CoreferenceResolver
Bidaf
...
```

I think this is easy to add - just check if the full path is a directory in allennlp and then list the Registerable classes and Types. Feel free to leave to another PR though.